plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -530,6 +528,52 @@ private static void handleStandardArgument(Props jobProps,
   }
 
   /**
+   * This method is used to get spark home from plugin's jobtype config.
+   * If spark.{sparkVersion}.home is set in commonprivate.properties/private.properties, then that will be returned.
+   * If spark.{sparkVersion}.home is not set and spark.home.dir is set then it will retrieve Spark directory inside
+   * spark.home.dir, matching spark.home.prefix + sparkVersion pattern. Spark directory name can be with dot
+   * in version name or without it. For e.g. spark-xxx-2_2105 or spark-xxx-2_2.1.0.5
+   * @param sparkVersion
+   * @return
+   */
+  private String getSparkHome(String sparkVersion) {
+    String sparkHome = getSysProps().get("spark." + sparkVersion + ".home");
+    if (sparkHome == null) {
+      info("Couldn't find spark." + sparkVersion + ".home property.");
+      String sparkDir = getSysProps().get("spark.home.dir");
+      String sparkHomePrefix =
+          getSysProps().get("spark.home.prefix") != null ? getSysProps().get("spark.home.prefix") : "*";
+      info(" Looking for spark at  " + sparkDir + " directory with " + sparkHomePrefix + " prefix for " + sparkVersion
+          + " version.");
+      DirectoryScanner scanner = new DirectoryScanner();
+      scanner.setBasedir(sparkDir);
+      scanner.setIncludes(
+          new String[]{sparkHomePrefix + sparkVersion.replace(".", "") + "*", sparkHomePrefix + sparkVersion + "*"});
+      scanner.scan();
+      String[] directories = scanner.getIncludedDirectories();
+      if (directories != null && directories.length > 0) {
+        sparkHome = sparkDir + "/" + directories[directories.length - 1];
+      } else {
+        scanner.setIncludes(new String[]{sparkHomePrefix + "*"});
+        scanner.scan();
+        String[] availableDirectories = scanner.getIncludedDirectories();
+        if (availableDirectories != null && availableDirectories.length > 0) {
+          Set<String> availableVersions = Sets.newTreeSet();
+          for (String directory : availableDirectories) {
+            String version = directory.replace(sparkHomePrefix, "");
+            availableVersions.add(version.contains(".") ? version.substring(0, 5)
+                : Joiner.on(".").join(version.substring(0, 3).split("")));
```

Comment:
We should change our Spark directory naming convention to prevent this possible confusion.
Before that, we could ignore this corner case since none of the currently deployed version has this problem.
That also brings another issue, is it possible to configure the naming convention in the config file so that we don't have to change binary each time the naming convention changes?